haystack_experimental/components/extractors/llm_metadata_extractor.py

```diff
@@ -248,37 +264,111 @@ def from_dict(cls, data: Dict[str, Any]) -> "LLMMetadataExtractor":
         """
 
         init_parameters = data.get("init_parameters", {})
+
         if "generator_api" in init_parameters:
             data["init_parameters"]["generator_api"] = LLMProvider.from_str(data["init_parameters"]["generator_api"])
+
         if "generator_api_params" in init_parameters:
-            deserialize_secrets_inplace(data["init_parameters"]["generator_api_params"], keys=["api_key"])
+            # Check all the keys that need to be deserialized
+            azure_openai_keys = ["azure_ad_token"]
+            aws_bedrock_keys = [
+                "aws_access_key_id", "aws_secret_access_key", "aws_session_token", "aws_region_name", "aws_profile_name"
+            ]
+            deserialize_secrets_inplace(
+                data["init_parameters"]["generator_api_params"],
+                keys=["api_key"] + azure_openai_keys + aws_bedrock_keys,
```

Comment:
don't you also need to add the OpenAI key here?

Reply:
The OpenAI key is called `api_key` in the init method so it's covered.